ansible_base/api_documentation/path_utils.py

```diff
@@ -0,0 +1,55 @@
+"""
+Utility functions for parsing API paths and operation IDs.
+
+This module provides shared utilities used by both preprocessing and
+postprocessing hooks to ensure consistent path parsing logic.
+"""
```

Comment:
Do any of these utilities have value outside of api_documentation? i.e. should we put this into ansible_base/lib/ somewhere so others can reuse/